experimental/MatroidRealizationSpaces/src/SelfProjectingMatroids.jl

```diff
@@ -0,0 +1,336 @@
+julia> M = fano_matroid();
+```
+"""
+
+#this function is not properly tested, since it did not terminate for intersting examples.
+function selfproj_realization_ideal(m::Matroid;saturate::Bool = false)::Ideal
+    @warn "This function is slow except for small matroids!"
+    if !is_selfprojecting(m) 
+      error("The given matroid is not self-projecting.") #Is it too costly to have this check?! How about an option to check, which can be turned off?
+    end
+    if !is_realizable(m,char = 0)
+      return defining_ideal(realization_space(m,char =0, QQ))
+    end
+    RS = realization_space(m,char =0,simplify = true, saturate = true); 
+    R = ambient_ring(RS); 
+    I = defining_ideal(RS);
+    n = length(matroid_groundset(m))
+    k = rank(m)
+    RR, x,l = polynomial_ring(QQ, :x=> 1:length(gens(R)), :l=>1:n; cached=false)
+    F = hom(R, RR,x) #use this to move ideals into RR
+    G = hom(RR, R, vcat(gens(R),[0 for i in 1:n])) #use this to move them back again into the ambient ring of RS #Careful! this sets all occurences of l to zero! Use only when l is eliminated !
+    L = ideal(RR, prod(l[i] for i in 1:n))
+    MRS = realization_matrix(RS)
+    M = matrix(RR,nrows(MRS),ncols(MRS),[F(MRS[i,j]) for i in 1:nrows(MRS) for j in 1:ncols(MRS)])
+    DL = diagonal_matrix(RR,[l[i] for i in 1:n])
+    V = M*DL*transpose(M)
+    IS = ideal(RR,[V[i,j] for i in 1:k for j in 1:k]) + ideal(RR,[F(gens(I)[i]) for i in 1:length(gens(I))])
+    J = stepwise_saturation(IS,l) 
+    E = eliminate(J,l) 
+    if saturate 
+      E = stepwise_saturation(E,F.(inequations(RS)))
+    end
+    if is_zero(E)
+      return ideal(ambient_ring(RS),[0]);
+    elseif is_one(E)
+      return ideal(ambient_ring(RS),[1]);
+    else
+      return ideal(R,G.(gens(E)))
+    end
+end
+@doc raw"""
+    selfproj_realization_ideal(m::Matroid)
+
+    Function to compute the defining_ideal of a selfprojecting realization space
+"""
+
+#this function computes the minors of a given matrix for the bases in a given list.
+#this list is not reduced, i.e. there are no repetitions but the polynomials are not reduced, i.e. the list might contain x+1 and y-2 and their product
+function sort_by_degree(polys::Vector{<:RingElem})::Vector{<:RingElem}
+    deg(p) = hasmethod(total_degree, Tuple{typeof(p)}) ? total_degree(p) : 0
+    return sort(polys, by=deg)
+end
+
+function basisminors(M::MatElem, Bases::Vector{Vector{Int}})::Vector{<:RingElem}
+    R = base_ring(M);
+    candidates = sort_by_degree([R(det(M[1:nrows(M),Bases[i]])) for i in 1:length(Bases)])
+    multiplicativeSet= nothing
+    ineqs = [R(0)]
+    if R isa MPolyQuoRing #in this case I cannot make the multiplicativeSet using the powers_of_element so we currently use the lesser choice: here there might be double entries or products of polynomials in the list - is there a better solution for this?
+      for i in 1:length(candidates)
+        if isone(candidates[i]) || isone(-candidates[i]) 
+        elseif iszero(candidates[i])
```

Comment:
you should use the @req macro in this case since it will through an argument error
```suggestion
        @req !iszero(candidates[i]) "a basis has vanishing minor"
```